buildSrc/src/main/kotlin/pklJavaLibrary.gradle.kts

```diff
@@ -80,9 +94,34 @@ val workAroundKotlinGradlePluginBug by
     }
   }
 
+val truffleJavacArgs =
+  listOf(
+    // TODO: determine correct limits for Truffle specializations
+    // (see https://graalvm.slack.com/archives/CNQSB2DHD/p1712380902746829)
+    "-Atruffle.dsl.SuppressWarnings=truffle-limit"
+  )
+
+val jpmsJavacArgs = listOf("--add-modules=jdk.unsupported")
+val javacArgsProvider = CommandLineArgumentProvider { jpmsJavacArgs }
+
 tasks.compileJava {
+  javaCompiler = info.javaCompiler
   dependsOn(workAroundKotlinGradlePluginBug)
-  // TODO: determine correct limits for Truffle specializations
-  // (see https://graalvm.slack.com/archives/CNQSB2DHD/p1712380902746829)
-  options.compilerArgs.add("-Atruffle.dsl.SuppressWarnings=truffle-limit")
+  options.compilerArgumentProviders.add(CommandLineArgumentProvider { truffleJavacArgs })
+}
+
+tasks.withType<JavaCompile>().configureEach {
+  val jvmTarget = JavaVersion.toVersion(info.jvmTarget)
+  javaCompiler = info.javaCompiler
+  sourceCompatibility = jvmTarget.majorVersion
+  targetCompatibility = jvmTarget.majorVersion
```

Comment:
bytecode levels are set explicitly so that any JVM17+ toolchain can be used without introducing compatibility risks with earlier JVM's/JDKs (particularly in downstream projects...)

Reply:
How come this is configuring both `tasks.compileJava` and `tasks.withType<JavaCompile>`?